### PR TITLE
Pickup notification creates shipment for default source only #2222.

### PR DESCRIPTION
**Pull request by novikor**
Set the pickup location source as shipping argument.

### Fixed Issues (if relevant)

1. magento-engcom/msi#2222: Pickup notification creates shipment for default source only
2. magento-engcom/msi#2164: Order Ready for Pickup tests coverage

### Manual testing scenarios (*)
0. Create a source with code `pickup` and mark it as used for pickup
1. Create an order
2. Mark it as placed using Store Pickup 
`INSERT IGNORE 
INTO inventory_pickup_location_order (order_id, pickup_location_code)
SELECT entity_id, 'pickup'
FROM sales_order;`
3. Go to order view page and press 'Notify Order is Ready for Pickup'

### Contribution checklist (*)
 - [x] Pull request has a meaningful description of its purpose
 - [x] All commits are accompanied by meaningful commit messages
 - [x] All new or changed code is covered with unit/integration tests (if applicable)
 - [x] All automated tests passed successfully (all builds on Travis CI are green)

### CODE REVIEW

**Comment by novikor** [comment]
I will cover Pickup Notification functionality by Integration tests one day.

File: app/code/Magento/InventoryInStorePickup/Model/NotifyOrderIsReadyForPickup.php [hunk 1]
```diff
@@ -66,11 +86,41 @@ public function execute(int $orderId): void
...
+
+    /**
+     * @param OrderInterface $order
+     *
+     * @return ShipmentCreationArgumentsInterface
+     */
+    private function getShipmentArguments(OrderInterface $order): ShipmentCreationArgumentsInterface
+    {
+        $arguments = $this->shipmentArgumentsFactory->create();
+        /* We have already checked that PickupLocationCode exists */
```
Comment (novikor) [inline app/code/Magento/InventoryInStorePickup/Model/NotifyOrderIsReadyForPickup.php:116]: Please don`t miss this line during the code review.

**Comment by ishakhsuvarov** [comment]
> I will cover Pickup Notification functionality by Integration tests one day.

The sooner the better. Hope that one day comes soon.

**Comment by novikor** [comment]
> > I will cover Pickup Notification functionality by Integration tests one day.
> 
> The sooner the better. Hope that one day comes soon.

We already have the issue for it. I will contact @vpodorozh or do it by myself this weekends

**Review by ishakhsuvarov — APPROVED** [review]
What about adding a test? Since it was broken at least once it's a good candidate.
Yes, I know, we discussed it. Just a friendly reminder 😄

**Comment by vpodorozh** [comment]
> What about adding a test? Since it was broken at least once it's a good candidate.
> Yes, I know, we discussed it. Just a friendly reminder 😄

@ishakhsuvarov - it is in progress now. Due to buggy functionality, I'm spending additional time to figure out how should it works before test coverage, and fix this the bugs that was found while test coverage :)
branch: `store-pickup-Order-Ready-for-Pickup-tests-coverage-IntegrationTests-#2164`
I'll create a PR for review till Sunday (19.05)!
PR for MFTF will be finished till Thuesday (21.05).

Thx!

**Comment by ishakhsuvarov** [comment]
That seems like a solid plan. Anyway, feel free to reach out if something is buggy or working no as expected.

**Comment by vpodorozh** [comment]
> I will cover Pickup Notification functionality by Integration tests one day.

PR: #2268

**Comment by novikor** [comment]
@ishakhsuvarov , the tests coverage have been delivered.

File: app/code/Magento/InventoryInStorePickup/Model/Order/IsFulfillable.php [hunk 2]
```diff
@@ -86,8 +97,11 @@ private function isItemFulfillable(string $sku, string $sourceCode, float $qtyOr
         if ($sourceItems->getTotalCount()) {
             /** @var SourceItemInterface $sourceItem */
             $sourceItem = current($sourceItems->getItems());
+            $source = $this->sourceRepository->get($sourceCode);
 
-            return bccomp((string)$sourceItem->getQuantity(), (string)$qtyOrdered) >= 0;
+            return bccomp((string)$sourceItem->getQuantity(), (string)$qtyOrdered, 1) >= 0 &&
```
Comment (ishakhsuvarov) [inline app/code/Magento/InventoryInStorePickup/Model/Order/IsFulfillable.php]: Why `bccomp` has scale of 1?

Comment (vpodorozh) [reply app/code/Magento/InventoryInStorePickup/Model/Order/IsFulfillable.php]: Fixed - now it is 4 :)

**Pull request merged**